### PR TITLE
Use new storage.Appender Append() instead of Add & AddFast

### LINKED ISSUE #3812: Experiment with single Append Prometheus method is blocking us or not.
Ref: https://github.com/prometheus/prometheus/pull/8489#issuecomment-781442995

### PR DESCRIPTION
**Pull request by metalmatze**
Closes #3812 

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

Update to latest Prometheus to use the new storage.Appener interface with `Append()` and removed `Add()` & `AddFast()`

## Verification

~Tests should pass for the most part - we still need to benchmark all of it!~ done!

### CODE REVIEW

**Comment by metalmatze** [comment]
Ran the new benchmark `BenchmarkMultiTSDB` (see PR) on both branches, main and this one.
```sh
 go test -bench=BenchmarkMultiTSDB -count=10 -benchtime=10000000x ./pkg/receive > /tmp/add.txt    #main
 go test -bench=BenchmarkMultiTSDB -count=10 -benchtime=10000000x ./pkg/receive > /tmp/append.txt #metalmatze:appender-append
```

Here are the benchstat results:
```
benchstat /tmp/add.txt /tmp/append.txt
name          old time/op    new time/op    delta
MultiTSDB-24     226ns ± 4%     219ns ± 4%  -2.85%  (p=0.003 n=10+9)

name          old alloc/op   new alloc/op   delta
MultiTSDB-24      171B ± 0%      171B ± 0%    ~     (all equal)

name          old allocs/op  new allocs/op  delta
MultiTSDB-24      0.00           0.00         ~     (all equal)
```

Looking at it from this benchmark nothing drastically changed, time got slightly better but probably normally not noticeable.

**Comment by bwplotka** [comment]
Is this non WIP, right?

File: pkg/receive/writer.go [hunk 1]
```diff
@@ -142,21 +142,17 @@ func (f *fakeAppendable) Appender(_ context.Context) (storage.Appender, error) {
...
-	addErr      func() error
-	addFastErr  func() error
+	appendErr   func() error
 	commitErr   func() error
 	rollbackErr func() error
 }
 
 var _ storage.Appender = &fakeAppender{}
 
 // TODO(kakkoyun): Linter - `addFastErr` always receives `nil`.
```
Comment (bwplotka) [inline pkg/receive/writer.go:152]:
```suggestion

```

File: pkg/store/storepb/testutil/series.go [hunk 2]
```diff
@@ -77,11 +77,12 @@ func CreateHeadWithSeries(t testing.TB, j int, opts HeadGenOptions) (*tsdb.Head,
 	app := h.Appender(context.Background())
 	for i := 0; i < opts.Series; i++ {
 		ts := int64(j*opts.Series*opts.SamplesPerSeries + i*opts.SamplesPerSeries)
-		ref, err := app.Add(labels.FromStrings("foo", "bar", "i", fmt.Sprintf("%07d%s", ts, LabelLongSuffix)), ts, opts.Random.Float64())
+		ref, err := app.Append(0, labels.FromStrings("foo", "bar", "i", fmt.Sprintf("%07d%s", ts, LabelLongSuffix)), ts, opts.Random.Float64())
```
Comment (bwplotka) [inline pkg/store/storepb/testutil/series.go:80]: 💪🏽

**Review by bwplotka — APPROVED** [review]
Sweet, let's go! I added one suggestion, but overall LGTM. Thanks!

**Comment by metalmatze** [comment]
Yup, not WIP anymore. I applied your suggestion and we're ready to merge!

**Review by bwplotka — APPROVED** [review]

**Review by squat — APPROVED** [review]
Nice!

**Pull request merged**